src/Features/CSharp/Portable/CodeRefactorings/DeclareAsNullable/DeclareAsNullableCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,181 @@
+                    type = local.Type;
+                    break;
+                case IPropertySymbol property:
+                    type = property.Type;
+                    break;
+                case IMethodSymbol method when method.IsDefinition:
+                    type = method.ReturnType;
+                    break;
+                case IFieldSymbol field:
+                    type = field.Type;
+                    break;
+                default:
+                    return false;
+            }
+
+            return type?.IsReferenceType == true;
+        }
+
+        private TypeSyntax TryGetTypeToFix(SyntaxNode node)
+        {
+            switch (node)
+            {
+                case ParameterSyntax parameter:
+                    return parameter.Type;
+
+                case VariableDeclaratorSyntax declarator:
+                    if (declarator.IsParentKind(SyntaxKind.VariableDeclaration))
+                    {
+                        var declaration = (VariableDeclarationSyntax)declarator.Parent;
+                        return declaration.Variables.Count == 1 ? declaration.Type : null;
+                    }
+                    return null;
+
+                case PropertyDeclarationSyntax property:
+                    return property.Type;
+
+                case MethodDeclarationSyntax method:
+                    if (method.Modifiers.Any(SyntaxKind.PartialKeyword))
+                    {
+                        return null;
+                    }
+                    return method.ReturnType;
+            }
+
+            return null;
+        }
+
+        private async Task<ISymbol> TryGetSymbolToFix(SyntaxNode root, TextSpan textSpan, Document document, CancellationToken cancellationToken)
+        {
+            var token = root.FindToken(textSpan.Start);
+
+            if (!token.IsKind(SyntaxKind.EqualsEqualsToken, SyntaxKind.ExclamationEqualsToken, SyntaxKind.NullKeyword))
+            {
+                return null;
+            }
+
+            BinaryExpressionSyntax equals;
+            if (token.Parent.IsKind(SyntaxKind.EqualsExpression, SyntaxKind.NotEqualsExpression))
+            {
+                equals = (BinaryExpressionSyntax)token.Parent;
+            }
+            else if (token.Parent.IsKind(SyntaxKind.NullLiteralExpression) && token.Parent.IsParentKind(SyntaxKind.EqualsExpression, SyntaxKind.NotEqualsExpression))
+            {
+                equals = (BinaryExpressionSyntax)token.Parent.Parent;
+            }
+            else
+            {
+                return null;
+            }
+
+            ExpressionSyntax value;
+            if (equals.Right.IsKind(SyntaxKind.NullLiteralExpression))
+            {
+                value = equals.Left;
+            }
+            else if (equals.Left.IsKind(SyntaxKind.NullLiteralExpression))
+            {
+                value = equals.Right;
+            }
+            else
+            {
+                return null;
+            }
+
```

Comment:
consider supporting `is null` and possibly even `(object)x == null` and ReferenceEquals